src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -48,18 +49,20 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
     {
         $resources = $this->resourcesResolver->getResources($requestConfiguration, $repository);
 
-        if ($resources instanceof Pagerfanta) {
+        $data = $resources instanceof ResourceGridView ? $resources->getData() : $resources;
```

Comment:
yep, but the "get" method needs to return ResourceGridView when $resources is an instance of ResourceGridView. So if I overwrite $resources var, the method would return Pagerfanta instance.